contracts/FIFOMarket.sol

```diff
@@ -65,25 +77,53 @@ contract FIFOMarket is
     );
     _queueHeadIndex = 0;
     _queueNextInsertIndex = 0;
+    priorityRestrictedThreshold = 0;
+    totalSupply = 0;
+    _grantRole(DEFAULT_ADMIN_ROLE, _msgSender());
+    _grantRole(ALLOWLIST_ROLE, _msgSender()); // todo, do this here or have to do it explicitly after intialization?
+  }
+
+  function setPriorityRestrictedThreshold(uint256 threshold)
+    external
+    onlyRole(DEFAULT_ADMIN_ROLE)
+  {
+    priorityRestrictedThreshold = threshold;
+    emit PriorityRestrictedThresholdSet(threshold);
   }
 
   function _queueLength() private view returns (uint256) {
     return _queueNextInsertIndex - _queueHeadIndex;
   }
 
-  // todo do we want this to report NRTs in either or wei? because removal token balances are denominated in wei
-  function numberOfNrtsInQueue() public view returns (uint256) {
+  function numberOfNrtsInQueueComputed() public view returns (uint256) {
```

Comment:
I don't see a good reason to use this class method in tandem with a class variable. They both appear to be keeping track of the total number of NRTs in the queue, without regard to whether they're restricted or not.

Reply:
Deriving the value each time is O(n) wrt the size of the queue so it's simply not an option to do every time.  I think we need to keep the running total.  I was thinking of leaving this computed version of the method on the contract in for now to use in tests and to be able to detect any discrepancy that might arise between the truth and the running total